### DISCUSSION TITLE
Versioning for Avalonia v11 Question

### DISCUSSION THREAD

**Discussion by DamianSuess** [AvaloniaCommunity/Prism.Avalonia#26] [Ideas]
As per Issue #25, there is a breaking change in Avalonia where the library references Prism.Avalonia will need to be upgraded. Any thoughts on how to approach moving forward with Avalonia v0.11? 

**To toss out a couple of suggestions:**

1. Move forward with v11.x and don't look back
    * Bumping Prism.Avalonia's version to v8.1.97.11
    * Examples:
        * `v8.1.97.11-preview.11.8`
        * `v8.1.97.1021` - where `.1021` is Avalonia v10.21
    * Future revisions of Prism.Avalonia can continue to follow the Prism version release numbers
2. Maintain backward compatibility, resulting with 2 NuGet packages (_v0.10 and v11.x_)

To be honest, I've found that pulling the bandaid off has been successful in most cases. This is taking into account, Prism Library is solid - _requiring little change over the months/years_ - and Avalonia will continue moving forward with releases, not looking back.

## NuGet Naming Restrictions

* NuGet packages use [System.Version](https://learn.microsoft.com/en-us/dotnet/api/system.version), `Major.Minor.Patch.Revision`
* https://learn.microsoft.com/en-us/nuget/concepts/package-versioning

**Comment by rabbitism** [comment]
While I was checking the latest release of original Prism, I am really surprised that prism quits .NET Foundation and UnityContainer discontinued.

**Reply by DamianSuess** [reply]
Unity was a surprise in deed! As for the NET Foundation, they had their reasons.

**Comment by yumikou** [comment]
Is it possible to support net461? Unfortunately, I still have some customers using Win7, and installation of netcore3.1 on it is too complicated.

**Reply by DamianSuess** [reply]
> Is it possible to support net461? Unfortunately, I still have some customers using Win7, and installation of netcore3.1 on it is too complicated.

Thank you for your request. At this time all efforts are towards keeping up with the latest releases. If you would like to create a PR and create a sample project to validate basic functionality, please do 🙏

**Comment by DamianSuess** [comment]
As it appears, Avalonia v0.10.x is still being maintained while v11.0.0 is still in the preview release phase. The concern is maintaining a version number that is compatible with both the Prism Library (i.e. `8.1.97`) and Avalonia (`11.0.0`), while being compatible with [Nuget Package Versioning](https://learn.microsoft.com/en-us/nuget/concepts/package-versioning) rules which follow [Semantic Versioning v2.0.0](https://semver.org/spec/v2.0.0.html).

> NuGet does accept `vW.X.Y.Z-suffix` (_Major.Minor.Patch.Build-Suffix_).

Historically, Avalonia only uses a flavor of `major.minor` for its releases. and Prism uses the full `major.minor.patch`. If both hold true, we can utilize Avalonia in the `build` zone + an incremental 5th place. For example `...11001` could be for Avalonia `v11.00` and our first patch.

Prism.Avalonia aims to align with the Prism library versioning system as it's base Semantic Versioning, the following below is being proposed.

## Consider the Following

The proposed numbering schema balances on Prism's vX.Y.Z and Avalonia's vX.Y numbering schemas.

| Prism.Avalonia | Prism | Avalonia | Details |
|-|-|-|-|
| v8.1.97.1021 | 8.1.97 | v0.10.21
| v8.1.97.11-preview.11.8 | 8.1.97 | 11.0.0-Preview8
| v8.1.97.11000-rc1.1 | | v11.0.0-RC1.1 | Release Candidate
| v8.1.97.11010 | | v11.0.1 | Avalonia's first revision
| v8.1.97.11012 | | v11.0.1 | Our maintenance build (`2`) on the first revision
